src/data/nat/factorial/double_factorial.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2023 Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jake Levinson
+-/
+import data.nat.factorial.basic
+import algebra.big_operators.order
+import tactic.ring
+/-!
+# Double factorials
+
+This file defines the double factorial,
+  `n ‼ := n * (n - 2) * (n - 4) * ...`.
+
+## Main declarations
+
+* `nat.double_factorial`: The double factorial.
+-/
+
+namespace nat
+
+/-- `nat.double_factorial n` is the double factorial of `n`. -/
+@[simp] def double_factorial : ℕ → ℕ
+| 0       := 1
+| 1       := 1
+| (k + 2) := (k + 2) * double_factorial k
+
+-- This notation is `\!!` not two !'s
+notation n `‼`:10000 := double_factorial n
```

Comment:
I picked the priority level here randomly. I just wanted it to have higher priority than multiplication, so we can write `(n + 2) * n ‼`.

Reply:
I think you can use `postfix` instead of `notation`. Probably good to copy what is done for `nat.factorial`.